pkg/lib/authenticationflow/flow.go

```diff
@@ -121,3 +121,45 @@ func FindCurrentFlowReference(flow *Flow) *FlowReference {
 	}, flow)
 	return ref
 }
+
+// FindStepByType recursively finds the first step in the flow that matches the given stepType
+// and returns its JSON pointer.
+func FindStepByType(flowRoot config.AuthenticationFlowObjectFlowRoot, stepType config.AuthenticationFlowStepType) *jsonpointer.T {
```

Comment:
It is better to return `(jsonpointer.T, bool)`, since a zero value of jsonpointer.T is a valid value. It points to the root of the JSON document.

Reply:
This was addressed in a later commit.